concordium-consensus/src/Concordium/Scheduler.hs

```diff
@@ -2866,19 +2866,31 @@ handleChainUpdate (WithMetadata{wmdData = ui@UpdateInstruction{..}, ..}, maybeVe
                                     SMintDistributionVersion1 -> checkSigAndEnqueue $ UVMintDistribution u
                                 RootUpdatePayload (RootKeysRootUpdate u) -> checkSigAndEnqueue $ UVRootKeys u
                                 RootUpdatePayload (Level1KeysRootUpdate u) -> checkSigAndEnqueue $ UVLevel1Keys u
-                                RootUpdatePayload (Level2KeysRootUpdate u) -> case sAuthorizationsVersionFor scpv of
+                                RootUpdatePayload (Level2KeysRootUpdate u) -> case sauv of
                                     SAuthorizationsVersion0 -> checkSigAndEnqueue $ UVLevel2Keys u
                                     SAuthorizationsVersion1 -> return $ TxInvalid NotSupportedAtCurrentProtocolVersion
-                                RootUpdatePayload (Level2KeysRootUpdateV1 u) -> case sAuthorizationsVersionFor scpv of
+                                    SAuthorizationsVersion2 -> return $ TxInvalid NotSupportedAtCurrentProtocolVersion
```

Comment:
I think the failure cases could be consolidated as `_ -> return $ TxInvalid NotSupportedAtCurrentProtocolVersion`, and similarly with the other cases below.